src/provider/mod.rs

```diff
@@ -241,6 +257,184 @@ where
             .request("zks_getProof", (address, keys, l1_batch_number))
             .into()
     }
+
+    /// Waits for the L1 -> L2 transaction to be included into the L2 block.
+    ///
+    /// ## Parameters
+    ///
+    /// - `l1_tx_receipt`: receipt of the transaction that initiated an L2 transaction request.
+    /// - `polling_interval_in_seconds`: Optional: polling interval for getting the L2 tx receipt in seconds.
+    /// - `timeout_in_seconds`: Optional: total timeout to wait in seconds.
+    ///
+    /// ## Returns
+    ///
+    /// L2 transaction receipt response.
+    async fn wait_for_l1_tx(
+        &self,
+        l1_tx_receipt: L1TransactionReceipt,
+        polling_interval_in_seconds: Option<Duration>,
+        timeout_in_seconds: Option<Duration>,
+    ) -> Result<ReceiptResponse, Error> {
+        let l1_to_l2_tx_log = l1_tx_receipt
+            .inner
+            .logs()
+            .iter()
+            .filter_map(|log| log.log_decode::<Bridgehub::NewPriorityRequest>().ok())
+            .next().ok_or(
```

Comment:
Nit: the formatting feels broken here, probably you need to move `.ok_or` to another line + break the error message into multiple lines.